packages/ui/core-components/src/lib/unsorted/viz/table/DataTable.stories.svelte

```diff
@@ -179,3 +181,27 @@
 	)}
 	<DataTable {data} link="link" showLinkCol />
 </Story>
+
+<Story name="Conditional Columns">
```

Comment:
@csjh Please add a `play` function with some assertions to test this functionality